modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/SimplePlayer.kt

```diff
@@ -59,6 +68,9 @@ class SimplePlayer(val settings: Settings, val statsManager: StatsManager, val c
 
     private var videoChangedListener: VideoChangedListener? = null
 
+    private var databaseProvider: StandaloneDatabaseProvider? = null
+    private var simpleCache: SimpleCache? = null
```

Comment:
I'm not sure but it might be simpler to make these properties `lazy` and then access them only when the feature flag is enabled instead of relying on `run()` and nullability.

Reply:
I was thinking of something like this.

```kotlin
private val databaseProvider by lazy { StandaloneDatabaseProvider(context) }
private val simpleCache by lazy {
    SimpleCache(
        File(context.cacheDir, "podcasts-cache"),
        LeastRecentlyUsedCacheEvictor(MAX_DEVICE_CACHE_SIZE_BYTES.toLong()),
        databaseProvider,
    )
}

val sourceFactory = if (FeatureFlag.isEnabled(Feature.CACHE_PLAYING_EPISODE) && location is EpisodeLocation.Stream) {
    CacheDataSource.Factory()
            .setCache(simpleCache)
            .setUpstreamDataSourceFactory(httpDataSourceFactory)
} else {
    dataSourceFactory
}

if (FeatureFlag.isEnabled(Feature.CACHE_PLAYING_EPISODE)) {
    databaseProvider.close()
    simpleCache.release()
}
```

Reply:
I would need to investigate deeper, but when I change to `lazy` I got this error after tapping on `play` button:

<img src="https://github.com/Automattic/pocket-casts-android/assets/42220351/51272f06-31ee-4261-84c8-f6c3a4a08a3f"  height="400">


I am getting this after applying this change. If I revert them it works as expected

Reply:
I like the change and it's simple, but for some reason it's not working as expected

Reply:
Ok, then let's ignore it and move on with your solution. 👍